src/js/game/LevelMVC/LevelModel.js

```diff
@@ -261,6 +261,46 @@ module.exports = class LevelModel {
     return this.controller.score;
   }
 
+  shouldRide(direction) {
+    let player = this.player;
+    let frontPosition = this.getNextRailPosition(player);
+    let frontBlock = this.actionPlane.getBlockAt(frontPosition);
+    return this.isNextRailValid(frontBlock, direction);
+  }
+
+  isNextRailValid(block, direction) {
+    if (!block) {
+      return;
+    }
+    return this.inverseDirection(block.connectionA) === direction ||
+    this.inverseDirection(block.connectionB) === direction ||
+    block.connectionA === direction ||
+    block.connectionB === direction;
+  }
+
+  inverseDirection(direction) {
+    return (direction + 2) % 4;
+  }
+
+  getNextRailPosition(entity = this.player) {
+    let offset = [0,0];
+    switch (entity.movementState) {
+      case 0:
+        offset[1] = -1;
+        break;
+      case 1:
+        offset[0] = 1;
+        break;
+      case 2:
+        offset[1] = 1;
+        break;
+      case 3:
+        offset[0] = -1;
+        break;
+    }
+    return [entity.position[0] + offset[0], entity.position[1] + offset[1]];
+  }
```

Comment:
This can all simply be `Position.add(entity.position, FacingDirection.directionToOffset(entity.movementState))`

Reply:
that as a singular statement won't always work because once the player is on the rails (past the first index), they're being moved due to the rail tracking logic, not a moveDirection call, which means it's possible for their movementState to be -1 which will return undefined from the directionToOffset call, which will cause a crash further down the line.

Using the switch statement above and calling:
return Position.add(entity.position, offset);

will mean that if movementState was -1 we'll simply fall through and offset will be [0,0] and everything works as expected. So this is what I've done 🙆‍♂️